src/main/java/se/iths/crimedatabase/security/SecurityConfig.java

```diff
@@ -11,22 +11,26 @@
 @Configuration
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
-
-    private final String[] urls = {"/addresses", "/categories", "/crimes", "/criminals", "/users", "/victims", "/publish"};
-
     @Override
     protected void configure(HttpSecurity http) throws Exception {
         http
-                .csrf()
-                .ignoringAntMatchers(urls)
+                .csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/images/**").permitAll()
+                .antMatchers("/showCrimes").hasRole("ADMIN")
+                .antMatchers("/showUsers").hasRole("ADMIN")
+                .antMatchers("/showCategories").hasRole("ADMIN")
+                .antMatchers("/showVictims").hasRole("ADMIN")
+                .antMatchers("/showCriminals").hasRole("ADMIN")
+                .antMatchers("/showAddresses").hasRole("ADMIN")
+                .anyRequest().authenticated()
                 .and()
-                .httpBasic()
+                .formLogin().usernameParameter("username")
```

Comment:
"username" is the default usernameParameter so it should work without ".usernameParameter("username")"